core/domain/exp_domain.py

```diff
@@ -655,6 +659,76 @@ def validate(self):
         self.outcome.validate()
 
 
+class Hint(object):
+    """Value object representing a hint."""
+    def __init__(self, hint_text, views):
+        self.hint_text = hint_text
+        self.views = views
+
+    def to_dict(self):
+        return {
+            'hint_text': self.hint_text,
+            'views': self.views,
+        }
+
+    @classmethod
+    def from_dict(cls, hint_dict):
+        return cls(
+            hint_dict['hint_text'],
+            hint_dict['views'])
+
+    def validate(self):
+        if not isinstance(self.hint_text, basestring):
+            raise utils.ValidationError(
+                'Expected hint text to be a string, received %s' %
+                self.hint_text)
+        if not isinstance(self.views, int):
+            raise utils.ValidationError(
+                'Expected views to be an integer, received %s' %
+                self.views)
+
+
+class Solution(object):
+    """Value object representing a solution.
+
+    A solution consists of answer_boolean, correct_answer and an explanation.
+    answer_boolean when True indicates that it is the only correct answer and
+    False indicates that it is one possible answer. correct_answer records the
+    answer to progress to the next card and explanation is a string containing
+    an explanation for the solution.
+    """
+    def __init__(self, answer_boolean, correct_answer, explanation):
+        self.answer_boolean = answer_boolean
+        self.correct_answer = correct_answer
+        self.explanation = explanation
+
+    def to_dict(self):
+        return {
+            'answer_boolean': self.answer_boolean,
+            'correct_answer': self.correct_answer,
+            'explanation': self.explanation,
```

Comment:
The explanation should be scrubbed. If the correct_answer contains arbitrary HTML it should also be scrubbed.